--- a/src/lxc/confile.c	2016-03-17 18:12:45.616074129 +0800
+++ b/src/lxc/confile.c	2016-03-17 18:13:03.328074562 +0800
@@ -2252,7 +2252,7 @@
 		return lxc_clear_config_keepcaps(c);
 	else if (strncmp(key, "lxc.cgroup", 10) == 0)
 		return lxc_clear_cgroups(c, key);
-	else if (strcmp(key, "lxc.mount.entries") == 0)
+	else if (strcmp(key, "lxc.mount.entry") == 0)
 		return lxc_clear_mount_entries(c);
 	else if (strcmp(key, "lxc.mount.auto") == 0)
 		return lxc_clear_automounts(c);
